plugin/noop.go

```diff
@@ -0,0 +1,166 @@
+package plugin
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/connylabs/ingest"
+	"github.com/connylabs/ingest/storage"
+)
+
+var (
+	// DefaultCodec can be used for testing against this noop plugin.
+	DefaultCodec  = ingest.SimpleCodec{XID: "id", XName: "name"}
+	DefaultLogger = hclog.New(&hclog.LoggerOptions{
```

Comment:
This is probably the only one that _should_ be kept exported so that other ingest plugins can make use of it?